### DISCUSSION TITLE
Click to move & pathfind

### DISCUSSION THREAD

**Discussion by salsa2k** [excaliburjs/Excalibur#2358] [Q&A] [answered]
Hi,

I have 3 questions, all related to actor movement.

1. Clicking to move an actor:

I'm trying to make a click to move but the movement is not going in the right direction.
```
  public onInitialize(engine: ex.Engine) {
    engine.input.pointers.primary.on('down', (evt) => this.MoveTo(evt));
    this.graphics.use(this.sprite);
  }

  public MoveTo(evt: ex.Input.PointerEvent) {
    const dir = evt.worldPos.add(this.pos);
    this.actions.moveTo(dir, this.playerSpeed);
  }
```

2. How do I stop an actor if I want to change the direction or simply stop the actor?

3. There is any pathfinder solution? I did not find anything in the documentation.

Thanks :)

**Comment by eonarheim** [comment, marked as answer]
Hi @salsa2k 

1. I think you can just use the `evt.worldPos` in your `moveTo` no need to do any arithmetic on the pointer event world position. `moveTo` let's you specify an absolute word position, `moveBy` would be relative to the current actors position👍 

```typescript
  public MoveTo(evt: ex.Input.PointerEvent) {
    this.actions.moveTo(evt.worldPos, this.playerSpeed);
  }
```

2. You can stop any running actions with the `this.actions.clearActions()` , I'll admit that method doesn't have the best name... perhaps "stop()" or "stopActions()"?

3. We don't have anything built in for pathfinding, but we have done A* pathfinding in a jam game before (warning old version of excalibur). In this example we build up a grid of positions that an actor can be in called "WaypointNodes"  and using A* to pathfind https://github.com/excaliburjs/ludum-38/blob/master/game/WaypointGrid.ts enemies could then wander to a specific node and build up a path to follow with actions https://github.com/excaliburjs/ludum-38/blob/master/game/Enemy.ts#L135
  * I want to pull this pathfinding into a standalone sample, I should have some time this week to do that 👍 

Hope this helps!

**Comment by florianjs** [comment]
I know that this discussion is old, but I'm trying to implement easystar (https://github.com/prettymuchbryce/easystarjs) 
Thing is, I need to convert worldPos into 'Grid Position'. Will keep you update

**Reply by eonarheim** [reply]
@florianjs Definitely keep us posted!

Take a look at what we did here, might help plug in easystar

https://github.com/excaliburjs/sample-tactics/tree/main/src/path-finding
